docs/principles.md

```diff
@@ -0,0 +1,69 @@
+# OmnICU Design Principles & Decisions
+
+## Guiding principles
+
+### Immutable objects
+
+Discussion about builders, constructors. But no object is to be mutated after creation.
+
+### No I/O in the core library
+
+Leave I/O to the wrapper or clients. Pass data as reference to the methods.
+Clients can mmap, async fetch, package data with the library...
+
+### No internal threading
+
+Both Rust and Wasm support multithreading but we don’t have a need for it in the i18n realm.
+
+To simplify our library, and make sure we don’t have cross platform/language compatibility issues, one should avoid using threads in the core library.
+
+### No global mutable data
```

Comment:
Are local memoizers ok? We use it for `fluent-rs` - https://github.com/projectfluent/fluent-rs/tree/master/intl-memoizer